client/components/ModalWithForm/index.jsx

```diff
@@ -24,7 +27,11 @@ export class Component extends React.Component {
     }
 
     submit() {
-        this.getFormInstance().submit();
+        this.setState({submitting: true});
+        this.getFormInstance().submit()
```

Comment:
In the `render` function, can we change `ref: 'form'` to not use strings ref but a function instead.

i.e. `ref: (node) => this.dom.form = node` and update all `this.refs.form` to be `this.dom.form` to remove deprecated string refs.